anndata/base.py

```diff
@@ -877,23 +772,27 @@ def _init_as_actual(
         # multi-dimensional array annotations
         # None or {} → empty arrays
         try:
-            if not isinstance(obsm, np.ndarray) and not obsm:
-                obsm = np.empty(self._n_obs, dtype=[])
-            if not isinstance(varm, np.ndarray) and not varm:
-                varm = np.empty(self._n_vars, dtype=[])
+            if obsm is None:
+                obsm = dict()
+            if varm is None:
+                varm = dict()
         except TypeError:
```

Comment:
I believe I kept the try except because some test was failing without it. Looks like that's been fixed.

I switch back and forth on whether I prefer `dict` or `{}`. I currently prefer `dict()` and `list()` to `{}` and `[]` since it's more consistent with other types. Plus `{}` seems ambiguous with set literals.

Reply:
I think it’s a matter of taste as the argument goes either way. If you have e.g. `{'item'}` and remove the `'item'` you get `{}` and suddenly have a dict instead of a set. But if you have `set()` and change it to `set('item')` you get `{'i', 't', 'e', 'm'}` which is also not what you wanted.

Fun fact: Guido said that if he could redo Python, he’d use `{:}` as the empty dict and `{}` as the empty set.